pulsar/producer_partition.go

```diff
@@ -148,6 +179,26 @@ func newPartitionProducer(client *client, topic string, options *ProducerOptions
 	return p, nil
 }
 
+func (p *partitionProducer) sheduleDataKeyUpdate() {
+	for t := range p.dataKeyTicker.C {
+		p.log.Infof("Refreshing data key :%v", t)
+		err := p.updateDataKey()
+		if err != nil {
+			p.log.Errorf("Error refreshing data key : %v", err)
+		}
+	}
+}
+
+func (p *partitionProducer) updateDataKey() error {
```

Comment:
Does this need to be thread safe? Also, can we move the key refreshing to another MR.

Reply:
I can move it to next MR. 
Just ask:  why this to be moved to another MR?

Reply:
done